app/post_metadata.go

```diff
@@ -211,18 +211,14 @@ func (a *App) getEmbedForPost(post *model.Post, firstLink string, isNewPost bool
 	if permalink != nil {
 		embed := &model.PostEmbed{Type: model.POST_EMBED_PERMALINK}
 
-		if !model.IsValidId(askingUserID) {
-			return nil, fmt.Errorf("cannot embed permalink preview because user id %q is not valid", askingUserID)
-		}
-
 		if permalink.PreviewPost != nil {
 			referencedChannel, err := a.GetChannel(permalink.PreviewPost.Post.ChannelId)
 			if err != nil && err.StatusCode != http.StatusNotFound {
 				return nil, err
 			}
 
 			// Only embed the referenced post into the Data field if the containing channel wasn't deleted and the requesting user has access to the post.
-			if referencedChannel != nil && a.HasPermissionToChannel(askingUserID, referencedChannel.Id, model.PERMISSION_READ_CHANNEL) || (referencedChannel.Type == model.CHANNEL_OPEN && a.HasPermissionToTeam(askingUserID, referencedChannel.TeamId, model.PERMISSION_READ_PUBLIC_CHANNEL)) {
+			if askingUserID == "" || (referencedChannel != nil && a.HasPermissionToReadChannel(askingUserID, referencedChannel)) {
```

Comment:
Why pass the empty string for new posts and updates? I think it might be safer to keep passing that since I think it gives us an extra layer of security by preventing us from potentially accidentally leaking some data if a user without permission to see a post is posting a permalink to it.

I don't expect us to hopefully leak stuff here, but I'm trying to make the default case that we don't send anything just in case something gets broken in the future.

Reply:
Good question. There are absolutely other ways to achieve the same thing. It stems from the edge case that you might post a link to a post that you yourself can't access but others in the channel can.

This method puts the referenced post in the metatdata and *then* `nil`s it if for those who don't have authz. Alternatively, we could *not* put the referenced post in the metadata and then *add* it for those who have authz.

I totally see your point, you're concerned that someone might come along and use the app method with a blank `askingUserID` and expect to have the results authorized to the current user?

Reply:
I've completely removed the knowledge of the "asking user" from this method. It now embeds the referenced posts' metadata (when appropriate) regardless of authz rules. A later "sanitization" method can optionally be invoked to invoke the authz rules for the given user.